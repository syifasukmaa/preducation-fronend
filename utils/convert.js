const formatToCurrency = (number) => {
  const formattedNumber = new Intl.NumberFormat('id-ID', {
    style: 'currency',
    currency: 'IDR',
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  }).format(number)

  return formattedNumber
}

const formatToDate = (date) => {
  const dates = new Date(date)

  const options = {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    timeZone: 'Asia/Jakarta',
  }

  const formattedDate = dates.toLocaleString('id-ID', options)

  return formattedDate
}

export { formatToCurrency, formatToDate }
