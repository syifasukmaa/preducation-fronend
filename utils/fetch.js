export const loginAdmin = async (username, password) => {
  try {
    const response = await fetch(`${process.env.API_URL}/auths/admin/login`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        username,
        password,
      }),
    })
    return response
  } catch (error) {
    console.log(error)
  }
}

export const createNewCourse = async (token, courseData) => {
  try {
    const response = await fetch(`${process.env.API_URL}/courses`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(courseData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const updateCourse = async (token, courseId, updatedCourseData) => {
  try {
    const response = await fetch(`${process.env.API_URL}/courses/${courseId}`, {
      method: 'PATCH',
      headers: {
        Authorization: `Bearer ${token}`,
      },
      body: updatedCourseData,
    })

    return response
  } catch (error) {
    console.error('Error updating the course:', error)
    throw error
  }
}

export const deleteCourse = async (token, courseId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/courses/${courseId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    })

    return response
  } catch (error) {
    console.error('Error updating the course:', error)
    throw error
  }
}

export const createNewChapter = async (token, chapterData, courseId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/chapters/?courseId=${courseId}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(chapterData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const updateChapter = async (token, chapterData, chapterId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/chapters/${chapterId}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(chapterData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }

};
export const deleteChapter = async (token, chapterId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/chapters/${chapterId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    });

    return response;
  } catch (error) {
    console.error('Error updating the course:', error);
    throw error;
  }
};

export const createNewVideo = async (token, videoData, Id) => {
  try {
    const response = await fetch(`${process.env.API_URL}/videos/?chapterId=${Id}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(videoData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const updateVideo = async (token, videoData, Id) => {
  try {
    const response = await fetch(`${process.env.API_URL}/videos/${Id}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(videoData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const deleteVideo = async (token, videoId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/videos/${videoId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    })

    return response
  } catch (error) {
    console.error('Error updating the course:', error)
    throw error
  }
}

export const resetPassword = async (passwordResetToken, password, confirmPassword) => {
  try {
    const response = await fetch(`${process.env.API_URL}/auths/reset-password/${passwordResetToken}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        password,
        confirmPassword,
      }),
    })

    return response
  } catch (error) {
    console.log('Error creating a new course:', error)
    throw error
  }
}
