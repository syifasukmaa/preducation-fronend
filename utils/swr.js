import useSWR from 'swr'

export const fetcher = async (url, token) => {
  const headers = {}
  if (token) {
    headers.Authorization = `Bearer ${token}`
  }
  const res = await fetch(url, {
    headers,
  })
  const data = await res.json()
  return data
}

export const useCourse = (token, id, category, title) => {
  let titleQuery = ''

  if (title) {
    titleQuery = title
  }

  const { data, isLoading, mutate, error } = useSWR(
    token && id
      ? [`${process.env.API_URL}/courses/${id}`, token]
      : [`${process.env.API_URL}/courses/?category=${category}&title=${titleQuery}`, token],
    ([url, token]) => fetcher(url, token)
  )
  return {
    course: data?.data,
    isLoading,
    mutate,
    error,
  }
}

export const useChapter = (token, id) => {
  const { data, isLoading, mutate, error } = useSWR(
    token ? [`${process.env.API_URL}/chapters/${id}`, token] : null,
    ([url, token]) => fetcher(url, token)
  )
  return {
    chapter: data?.data,
    isLoading,
    mutate,
    error,
  }
}

export const usePayment = (token, status, username, limit, page) => {
  let statusQuery = ''
  let usernameQuery = ''
  let limitQuery = ''
  let pageQuery = ''
  if (status) {
    statusQuery = status
  }
  if (username) {
    usernameQuery = username
  }
  if (limit) {
    limitQuery = limit
  }
  if (page) {
    pageQuery = page
  }

  const { data, isLoading, mutate, error } = useSWR(
    token
      ? [
          `${process.env.API_URL}/payments/all/?status=${statusQuery}&username=${usernameQuery}&limit=${limitQuery}&page=${pageQuery}`,
          token,
        ]
      : null,
    ([url, token]) => fetcher(url, token)
  )
  return {
    payment: data?.data.payments,
    totalData: data?.data.total,
    isLoading,
    mutate,
    error,
  }
}

export const useCategory = (token, statistik) => {
  const { data, isLoading, mutate, error } = useSWR(
    token ? [`${process.env.API_URL}/categories/${statistik ? 'statistik' : ''}`, token] : null,
    ([url, token]) => fetcher(url, token)
  )
  return {
    categories: data?.data,
    isLoading,
    mutate,
    error,
  }
}
