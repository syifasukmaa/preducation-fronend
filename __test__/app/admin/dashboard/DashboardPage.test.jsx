import { render, fireEvent, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import DashboardPage from '@/app/admin/dashboard/page'
import * as nextAuthReact from 'next-auth/react'
import * as swrHook from '@/utils/swr'
import Layout, { metadata } from '@/app/admin/dashboard/layout'

jest.mock('next-auth/react', () => ({
  useSession: jest.fn(),
}))

jest.mock('@/utils/swr', () => ({
  usePayment: jest.fn(),
}))

describe('Dashboard Page Component', () => {
  let useSessionSpy
  let usePaymentSpy

  beforeEach(() => {
    useSessionSpy = jest.spyOn(nextAuthReact, 'useSession')
    useSessionSpy.mockReturnValue({
      data: {
        user: {
          accessToken: 'mockAccessToken',
        },
      },
    })

    usePaymentSpy = jest.spyOn(swrHook, 'usePayment')
    usePaymentSpy.mockReturnValue({
      payment: [
        {
          _id: '1',
          userId: { username: 'testUser' },
          courseId: { category: { name: 'TestCategory' }, level: 'TestLevel' },
          status: 'Paid',
          paymentType: 'Credit Card',
          createdAt: '2023-01-01T12:00:00Z',
        },
        {
          _id: '2',
          userId: { username: 'testUser2' },
          courseId: { category: { name: 'TestCategory' }, level: 'TestLevel' },
          status: 'On Progress',
          paymentType: 'Credit Card',
          createdAt: '2023-01-01T12:00:00Z',
        },
      ],
      isLoading: false,
      error: null,
      mutate: jest.fn(),
    })
  })

  afterEach(() => {
    useSessionSpy.mockRestore()
    usePaymentSpy.mockRestore()
  })
  it('renders the Dashboard page', async () => {
    const page = render(
      <Layout>
        <DashboardPage />
      </Layout>
    )
    expect(page).toMatchSnapshot()
    expect(page.getByText(/Status Pembayaran/i)).toBeInTheDocument()
  })
  it('should metadata is correct', () => {
    expect(metadata).toEqual({
      title: 'Preducation | Dashboard',
      description: 'Preducation online course',
    })
  })
  it('handles filter button click', async () => {
    const { container, getByTestId } = render(<DashboardPage />)

    fireEvent.click(getByTestId('filter-button'))
    expect(getByTestId('filter-popup-container')).toBeInTheDocument()

    fireEvent.click(getByTestId('close-button'))
    expect(container).not
      .toContain(`<div className="px-4 py-4 bg-white rounded-lg shadow-xl" data-testid="filter-popup-container">
        {children}
        <ClosesButton style={'absolute top-2 right-2 text-orange-05 hover:text-black'} onClick={clickClose} />
      </div>`)
  })

  it('handles search button click', async () => {
    const { container, getByTestId, getByPlaceholderText } = render(<DashboardPage />)

    fireEvent.click(getByTestId('search-button'))
    expect(getByPlaceholderText('Cari...')).toBeInTheDocument()
    expect(getByTestId('search-popup-container')).toBeInTheDocument()

    fireEvent.click(getByTestId('close-button'))
    expect(container).not
      .toContain(`<div data-testid="search-popup-container" className="absolute bg-white z-10 top-0 right-0">
      <div className="flex">
        <input
          type="text"
          placeholder="Cari..."
          className="border solid border-orange-05 px-6 py-1 rounded-md"
          value={title}
          onChange={(e) => setTitle(e.target.value)}
        />
        <ClosesButton onClick={onClick} style={'bg-orange-05 ml-1 text-white px-3 py-1 rounded'} />
      </div>
    </div>`)
  })

  it('ensures that filterCourses is called correctly when SUDAH BAYAR item is clicked', () => {
    const { getByTestId, getByText } = render(<DashboardPage />)

    fireEvent.click(getByTestId('filter-button'))
    fireEvent.click(getByTestId('paid-button'))
    expect(getByText('testUser2')).toBeInTheDocument()
  })

  it('ensures that filterCourses is called correctly when BELUM BAYAR item is clicked', () => {
    const { getByTestId, getByText } = render(<DashboardPage />)

    fireEvent.click(getByTestId('filter-button'))
    fireEvent.click(getByTestId('onprogress-button'))
    expect(getByText('testUser')).toBeInTheDocument()
  })

  it('handles refresh button click', async () => {
    const { getByTestId } = render(<DashboardPage />)

    fireEvent.click(getByTestId('refresh-button'))
    expect(usePaymentSpy.mock.calls[1][2]).toBe('')
  })

  it('handles outside click to close filter popup', async () => {
    const { getByTestId, queryByTestId } = render(<DashboardPage />)
    userEvent.click(getByTestId('filter-button'))
    userEvent.click(document.body)

    await waitFor(() => {
      expect(queryByTestId('filter-popup-container')).not.toBeInTheDocument()
    })
  })
  it('displays loading state', async () => {
    usePaymentSpy.mockReturnValue({ payment: null, isLoading: true, error: null, mutate: jest.fn() })

    const { getByTestId } = render(<DashboardPage />)
    expect(getByTestId('payment-loading1')).toBeInTheDocument()
  })
})
