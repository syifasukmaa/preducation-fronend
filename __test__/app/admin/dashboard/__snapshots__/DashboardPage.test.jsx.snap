// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Page Component renders the Dashboard page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="md:px-12 px-4"
      >
        <div
          class="relative flex items-center justify-between md:pt-2"
        >
          <p
            class="text-xl font-bold text-primary-dark-blue"
          >
            Status Pembayaran
          </p>
          <div
            class="relative flex items-center"
          >
            <svg
              class="mr-3 cursor-pointer text-orange-05 cursorPointer"
              data-testid="refresh-button"
              fill="none"
              height="25"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 12a9 9 0 0 0-9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
              />
              <path
                d="M3 3v5h5"
              />
              <path
                d="M3 12a9 9 0 0 0 9 9 9.75 9.75 0 0 0 6.74-2.74L21 16"
              />
              <path
                d="M16 16h5v5"
              />
            </svg>
            <button
              class="flex items-center border-[1px] border-orange-05 py-[2px] px-[6px] rounded-2xl mr-3 text-base hover:scale-105 transition-all"
              data-testid="filter-button"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 18 18"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_3248_770)"
                >
                  <path
                    clip-rule="evenodd"
                    d="M0.666748 1.49984C0.666748 1.0396 1.03984 0.666504 1.50008 0.666504H16.5001C16.9603 0.666504 17.3334 1.0396 17.3334 1.49984V3.65466C17.3334 4.09669 17.1578 4.52061 16.8453 4.83317L11.5001 10.1783V14.8332C11.5001 15.1488 11.3217 15.4374 11.0394 15.5785L7.70609 17.2452C7.44777 17.3744 7.14099 17.3606 6.89531 17.2087C6.64963 17.0569 6.50008 16.7887 6.50008 16.4998V10.1783L1.1549 4.83317C0.842342 4.52061 0.666748 4.09669 0.666748 3.65466V1.49984ZM2.33341 2.33317V3.65466L7.92267 9.24392C8.07895 9.4002 8.16675 9.61216 8.16675 9.83317V15.1515L9.83342 14.3181V9.83317C9.83342 9.61216 9.92121 9.4002 10.0775 9.24392L15.6667 3.65466V2.33317H2.33341Z"
                    fill="#EB5437"
                    fill-rule="evenodd"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_3248_770"
                  >
                    <rect
                      fill="white"
                      height="16.6667"
                      transform="translate(0.666748 0.666504)"
                      width="16.6667"
                    />
                  </clippath>
                </defs>
              </svg>
              <p
                class="ml-1 text-orange-05 font-bold text-base"
              >
                Filter
              </p>
            </button>
            <button
              aria-label="search"
              class="hover:scale-110 transition-all"
              data-testid="search-button"
            >
              <svg
                fill="none"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22.1333 24L13.7333 15.6C13.0667 16.1333 12.3 16.5556 11.4333 16.8667C10.5667 17.1778 9.64444 17.3333 8.66667 17.3333C6.24444 17.3333 4.19467 16.4942 2.51733 14.816C0.84 13.1378 0.000888889 11.088 0 8.66667C0 6.24444 0.839111 4.19467 2.51733 2.51733C4.19556 0.84 6.24533 0.000888889 8.66667 0C11.0889 0 13.1387 0.839111 14.816 2.51733C16.4933 4.19556 17.3324 6.24533 17.3333 8.66667C17.3333 9.64444 17.1778 10.5667 16.8667 11.4333C16.5556 12.3 16.1333 13.0667 15.6 13.7333L24 22.1333L22.1333 24ZM8.66667 14.6667C10.3333 14.6667 11.7502 14.0831 12.9173 12.916C14.0844 11.7489 14.6676 10.3324 14.6667 8.66667C14.6667 7 14.0831 5.58311 12.916 4.416C11.7489 3.24889 10.3324 2.66578 8.66667 2.66667C7 2.66667 5.58311 3.25022 4.416 4.41733C3.24889 5.58444 2.66578 7.00089 2.66667 8.66667C2.66667 10.3333 3.25022 11.7502 4.41733 12.9173C5.58444 14.0844 7.00089 14.6676 8.66667 14.6667Z"
                  fill="#EB5437"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="mt-4 mb-24 overflow-x-auto lg:mb-32 md:mt-6"
        >
          <div
            class="overflow-y-auto"
          >
            <table
              class="min-w-full bg-white rounded-lg"
            >
              <thead
                class="text-sm font-semibold bg-orange-04 text-neutral-05"
              >
                <tr>
                  <td
                    class="w-24 px-4 py-3"
                  >
                    ID
                  </td>
                  <td
                    class="w-32 px-4 py-3"
                  >
                    Kategori
                  </td>
                  <td
                    class="px-4 py-3"
                  >
                    Kelas Premium
                  </td>
                  <td
                    class="px-4 py-3"
                  >
                    Status
                  </td>
                  <td
                    class="px-4 py-3 lg:pl-4 lg:pr-0"
                  >
                    Metode Pembayaran
                  </td>
                  <td
                    class="px-4 py-3 pl-4 lg:pl-0 lg:pr-1"
                  >
                    Tanggal Bayar
                  </td>
                </tr>
              </thead>
              <tbody
                class="text-gray-700 whitespace-nowrap text-[10px]"
              >
                <tr>
                  <td
                    class="px-4 py-4 text-xs font-bold text-gray-05 w-[15%]"
                  >
                    testUser
                  </td>
                  <td
                    class="py-3 pl-4 pr-3 text-xs font-bold text-gray-05 w-[17%]"
                  >
                    TestCategory
                  </td>
                  <td
                    class="px-4 py-3 text-xs font-bold text-gray-04 w-[15%]"
                  >
                    TestLevel
                  </td>
                  <td
                    class="py-3 px-4 text-xs font-bold w-[15%] text-alert-green"
                  >
                    SUDAH BAYAR
                  </td>
                  <td
                    class="px-4 py-3 text-xs font-bold lg:pl-4 lg:pr-0 text-gray-04 w-[20%]"
                  >
                    Credit Card
                  </td>
                  <td
                    class="px-4 py-3 pl-4 text-xs font-bold lg:pl-0 lg:pr-1 text-gray-05"
                  >
                    1 Januari 2023 pukul 19.00.00
                  </td>
                </tr>
              </tbody>
              <tbody
                class="text-gray-700 whitespace-nowrap text-[10px]"
              >
                <tr>
                  <td
                    class="px-4 py-4 text-xs font-bold text-gray-05 w-[15%]"
                  >
                    testUser2
                  </td>
                  <td
                    class="py-3 pl-4 pr-3 text-xs font-bold text-gray-05 w-[17%]"
                  >
                    TestCategory
                  </td>
                  <td
                    class="px-4 py-3 text-xs font-bold text-gray-04 w-[15%]"
                  >
                    TestLevel
                  </td>
                  <td
                    class="py-3 px-4 text-xs font-bold w-[15%] text-alert-red"
                  >
                    BELUM BAYAR
                  </td>
                  <td
                    class="px-4 py-3 text-xs font-bold lg:pl-4 lg:pr-0 text-gray-04 w-[20%]"
                  >
                    Credit Card
                  </td>
                  <td
                    class="px-4 py-3 pl-4 text-xs font-bold lg:pl-0 lg:pr-1 text-gray-05"
                  >
                    1 Januari 2023 pukul 19.00.00
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="md:px-12 px-4"
    >
      <div
        class="relative flex items-center justify-between md:pt-2"
      >
        <p
          class="text-xl font-bold text-primary-dark-blue"
        >
          Status Pembayaran
        </p>
        <div
          class="relative flex items-center"
        >
          <svg
            class="mr-3 cursor-pointer text-orange-05 cursorPointer"
            data-testid="refresh-button"
            fill="none"
            height="25"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 12a9 9 0 0 0-9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
            />
            <path
              d="M3 3v5h5"
            />
            <path
              d="M3 12a9 9 0 0 0 9 9 9.75 9.75 0 0 0 6.74-2.74L21 16"
            />
            <path
              d="M16 16h5v5"
            />
          </svg>
          <button
            class="flex items-center border-[1px] border-orange-05 py-[2px] px-[6px] rounded-2xl mr-3 text-base hover:scale-105 transition-all"
            data-testid="filter-button"
          >
            <svg
              fill="none"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_3248_770)"
              >
                <path
                  clip-rule="evenodd"
                  d="M0.666748 1.49984C0.666748 1.0396 1.03984 0.666504 1.50008 0.666504H16.5001C16.9603 0.666504 17.3334 1.0396 17.3334 1.49984V3.65466C17.3334 4.09669 17.1578 4.52061 16.8453 4.83317L11.5001 10.1783V14.8332C11.5001 15.1488 11.3217 15.4374 11.0394 15.5785L7.70609 17.2452C7.44777 17.3744 7.14099 17.3606 6.89531 17.2087C6.64963 17.0569 6.50008 16.7887 6.50008 16.4998V10.1783L1.1549 4.83317C0.842342 4.52061 0.666748 4.09669 0.666748 3.65466V1.49984ZM2.33341 2.33317V3.65466L7.92267 9.24392C8.07895 9.4002 8.16675 9.61216 8.16675 9.83317V15.1515L9.83342 14.3181V9.83317C9.83342 9.61216 9.92121 9.4002 10.0775 9.24392L15.6667 3.65466V2.33317H2.33341Z"
                  fill="#EB5437"
                  fill-rule="evenodd"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_3248_770"
                >
                  <rect
                    fill="white"
                    height="16.6667"
                    transform="translate(0.666748 0.666504)"
                    width="16.6667"
                  />
                </clippath>
              </defs>
            </svg>
            <p
              class="ml-1 text-orange-05 font-bold text-base"
            >
              Filter
            </p>
          </button>
          <button
            aria-label="search"
            class="hover:scale-110 transition-all"
            data-testid="search-button"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22.1333 24L13.7333 15.6C13.0667 16.1333 12.3 16.5556 11.4333 16.8667C10.5667 17.1778 9.64444 17.3333 8.66667 17.3333C6.24444 17.3333 4.19467 16.4942 2.51733 14.816C0.84 13.1378 0.000888889 11.088 0 8.66667C0 6.24444 0.839111 4.19467 2.51733 2.51733C4.19556 0.84 6.24533 0.000888889 8.66667 0C11.0889 0 13.1387 0.839111 14.816 2.51733C16.4933 4.19556 17.3324 6.24533 17.3333 8.66667C17.3333 9.64444 17.1778 10.5667 16.8667 11.4333C16.5556 12.3 16.1333 13.0667 15.6 13.7333L24 22.1333L22.1333 24ZM8.66667 14.6667C10.3333 14.6667 11.7502 14.0831 12.9173 12.916C14.0844 11.7489 14.6676 10.3324 14.6667 8.66667C14.6667 7 14.0831 5.58311 12.916 4.416C11.7489 3.24889 10.3324 2.66578 8.66667 2.66667C7 2.66667 5.58311 3.25022 4.416 4.41733C3.24889 5.58444 2.66578 7.00089 2.66667 8.66667C2.66667 10.3333 3.25022 11.7502 4.41733 12.9173C5.58444 14.0844 7.00089 14.6676 8.66667 14.6667Z"
                fill="#EB5437"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="mt-4 mb-24 overflow-x-auto lg:mb-32 md:mt-6"
      >
        <div
          class="overflow-y-auto"
        >
          <table
            class="min-w-full bg-white rounded-lg"
          >
            <thead
              class="text-sm font-semibold bg-orange-04 text-neutral-05"
            >
              <tr>
                <td
                  class="w-24 px-4 py-3"
                >
                  ID
                </td>
                <td
                  class="w-32 px-4 py-3"
                >
                  Kategori
                </td>
                <td
                  class="px-4 py-3"
                >
                  Kelas Premium
                </td>
                <td
                  class="px-4 py-3"
                >
                  Status
                </td>
                <td
                  class="px-4 py-3 lg:pl-4 lg:pr-0"
                >
                  Metode Pembayaran
                </td>
                <td
                  class="px-4 py-3 pl-4 lg:pl-0 lg:pr-1"
                >
                  Tanggal Bayar
                </td>
              </tr>
            </thead>
            <tbody
              class="text-gray-700 whitespace-nowrap text-[10px]"
            >
              <tr>
                <td
                  class="px-4 py-4 text-xs font-bold text-gray-05 w-[15%]"
                >
                  testUser
                </td>
                <td
                  class="py-3 pl-4 pr-3 text-xs font-bold text-gray-05 w-[17%]"
                >
                  TestCategory
                </td>
                <td
                  class="px-4 py-3 text-xs font-bold text-gray-04 w-[15%]"
                >
                  TestLevel
                </td>
                <td
                  class="py-3 px-4 text-xs font-bold w-[15%] text-alert-green"
                >
                  SUDAH BAYAR
                </td>
                <td
                  class="px-4 py-3 text-xs font-bold lg:pl-4 lg:pr-0 text-gray-04 w-[20%]"
                >
                  Credit Card
                </td>
                <td
                  class="px-4 py-3 pl-4 text-xs font-bold lg:pl-0 lg:pr-1 text-gray-05"
                >
                  1 Januari 2023 pukul 19.00.00
                </td>
              </tr>
            </tbody>
            <tbody
              class="text-gray-700 whitespace-nowrap text-[10px]"
            >
              <tr>
                <td
                  class="px-4 py-4 text-xs font-bold text-gray-05 w-[15%]"
                >
                  testUser2
                </td>
                <td
                  class="py-3 pl-4 pr-3 text-xs font-bold text-gray-05 w-[17%]"
                >
                  TestCategory
                </td>
                <td
                  class="px-4 py-3 text-xs font-bold text-gray-04 w-[15%]"
                >
                  TestLevel
                </td>
                <td
                  class="py-3 px-4 text-xs font-bold w-[15%] text-alert-red"
                >
                  BELUM BAYAR
                </td>
                <td
                  class="px-4 py-3 text-xs font-bold lg:pl-4 lg:pr-0 text-gray-04 w-[20%]"
                >
                  Credit Card
                </td>
                <td
                  class="px-4 py-3 pl-4 text-xs font-bold lg:pl-0 lg:pr-1 text-gray-05"
                >
                  1 Januari 2023 pukul 19.00.00
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
